examples/mnist/mnist.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+import (
+	torch "github.com/wangkuiyi/gotorch"
+)
+
+func main() {
+	dataset := torch.NewMnist("./data")
+	dataset.AddTransforms([]torch.Transform{
+		torch.NewNormalize(0.1307, 0.3081),
+		torch.NewStack(),
+	})
+
+	//TODO(yancey1989): implement torch.DataLoader
+	//trainDataLoader := torch.NewDataLoader(dataset, 8)
+	//for batch_idx, batch := range trainDataLoader() {
```

Comment:
We need to have a Go way to write this loop structure so that Go+ can generate a Go program that follows this way.

We could follow how Go Authors would do with this loop. For example, https://golang.org/pkg/text/scanner/

Reply:
Done and updated the issue #30